src/machines/workflow_machines.rs

```diff
@@ -44,7 +48,7 @@ pub(crate) struct WorkflowMachines {
     replaying: bool,
     /// Workflow identifier
     pub workflow_id: String,
-    /// Identifies the current run and is used as a seed for faux-randomness.
+    /// Identifies the current run
     pub run_id: String,
```

Comment:
So, since we now have an expectation later on that this is in fact a well formed UUID, we should probably represent it as one here and make the `WorkflowMachines` constructor do it. The constructor is already fallible, so that deals with the parsing unwrap problem

Reply:
I've updated uuid_to_randomness_seed to return Result.

Reply:
@mfateev I think it'd be nice to have `WorkflowMachines.run_id` be a `UUID` -- and handle the error any time it's being set or in the constructor. Then `uuid_to_randomness_seed` can be infallible.

We might want to make an error type specific to this as well

Reply:
It would be weird to parse and stringify it all the time. The run_id is just passed around as a string.

Reply:
The reason would be that we're just delaying the inevitable -- if we know that `run_id` is supposed to be a well-formed UUID, then we should check that sooner rather than later. Crashing when we need to use it as a seed is incidental, if it was supposed to be well-formed all along. If it *doesn't* actually need to be well-formed, then that's maybe different, but it doesn't sound like that's true.